src/Transports/MassTransit.KafkaIntegration/Transport/KafkaRider.cs

```diff
@@ -36,38 +31,27 @@ public class KafkaRider :
 
             var topicAddress = new KafkaTopicAddress(_hostAddress, address);
 
-            KafkaProducerFactory<TKey, TValue> factory = GetProducerFactory<TKey, TValue>(topicAddress);
+            IKafkaProducerFactory<TKey, TValue> factory = GetProducerFactory<TKey, TValue>(topicAddress);
 
             return factory.CreateProducer(consumeContext);
         }
 
-        KafkaProducerFactory<TKey, TValue> GetProducerFactory<TKey, TValue>(Uri topicAddress)
+        IKafkaProducerFactory<TKey, TValue> GetProducerFactory<TKey, TValue>(Uri topicAddress)
             where TValue : class
         {
             if (!_producerFactories.TryGetValue(topicAddress, out var factory))
                 throw new ConfigurationException($"Producer for topic: {topicAddress} is not configured.");
 
-            if (factory is KafkaProducerFactory<TKey, TValue> producerFactory)
+            if (factory is IKafkaProducerFactory<TKey, TValue> producerFactory)
                 return producerFactory;
 
             throw new ConfigurationException($"Producer for topic: {topicAddress} is not configured for ${typeof(Message<TKey, TValue>).Name} message");
         }
 
-        protected override Task StartRider(CancellationToken cancellationToken)
+        protected override void AddReceiveEndpoint(IHost host)
```

Comment:
This should be using the host configuration to add receive endpoints, instead of using the Host (as discussed).

Reply:
Who calls this method? And why is it AddReceiveEndpoint? This concerns me.